src/ServiceInsight/MessageFlow/MessageNode.cs

```diff
@@ -39,20 +48,27 @@ public MessageNode(MessageFlowViewModel owner, StoredMessage message)
             });
         }
 
-        string ProcessSagaType(StoredMessage message)
+        IEnumerable<SagaInfo> ProcessSagaType(StoredMessage message)
         {
             if (message.Sagas == null)
             {
-                return string.Empty;
+                yield break;
             }
 
-            var originatingSaga = message.Sagas.FirstOrDefault();
-            if (originatingSaga == null)
+            foreach (var saga in message.Sagas)
             {
-                return string.Empty;
+                var originatingSaga = saga;
+                if (originatingSaga != null)
+                {
+                    yield return new SagaInfo
+                    {
+                        ID = saga.SagaId,
+                        SagaType = TypeHumanizer.ToName(originatingSaga.SagaType),
+                        IsSagaCompleted = saga.ChangeStatus == "Completed",
+                        IsSagaInitiated = string.IsNullOrEmpty(Message.GetHeaderByKey(MessageHeaderKeys.SagaId)) && !string.IsNullOrEmpty(Message.GetHeaderByKey(MessageHeaderKeys.OriginatedSagaId))
```

Comment:
Is it worth pushing the logic for these two down into `SagaInfo`

Reply:
yeah, good idea.